tests/core/test_model_manager.py

```diff
@@ -675,8 +681,9 @@ def test_train_by_module_class(reset_globals):
     with setup_test_trainer():
         train_future = caikit.train(SampleModule, DataStream.from_iterable([]))
         assert isinstance(train_future, ModelTrainerBase.ModelFutureBase)
-        loaded_model = train_future.load()
-        assert isinstance(loaded_model, SampleModule)
+        # ensure model actually trained
+        trained_model = train_future._trained_model
```

Comment:
I wonder if we should make a non-interface function on the `LocalModelTrainer` for this? We could then `assert isinstance(trainer, LocalModelTrainer)` and safely call a non-impl function (instead of relying on this hidden property)

Reply:
I don't think this is the `LocalModelTrainer`, it's a test class

Reply:
`TestModelFuture` in helpers.py